src/search/cegar/abstraction.cc

```diff
@@ -137,8 +137,8 @@ pair<int, int> Abstraction::refine(
 }
 
 void Abstraction::print_statistics() const {
```

Comment:
always log this or add an llog.is_at_least_normal?

Reply:
Currently, this is only called from `cegar.cc`, where it is ensured that logging is at least normal if calling the method. Are you in favor of always having that if-clause?

Reply:
For other functions I observed what you described (that the if is called around the function call). There I did not write a comment, because it was clear that the purpose of the function is just to pull the output generation away from the meaty parts of the function. To be honest, the function is called `print_statistics` I am fine with defining that if I call this function, then the statistics are printed, independent of the logging level.

Reply:
I agree that checking the log level is the responsibility of the caller.